blog/features/deprecating-the-service-worker.md

```diff
@@ -0,0 +1,254 @@
+
+### Web2 Developer Experience
+
+#### Reduced Cognitive Load
+
+There are a number of situations where the Service Worker cannot support Internet Computer developers and they must manually instead use ICX Proxy:
+
+- [SEO](https://developers.google.com/search/docs/fundamentals/seo-starter-guide) crawlers
+- Social media rich previews
+- Service workers & web workers
+- Home screen icons on iOS
+- Rich previews
+- Loading static assets from the Internet Computer into a web2 application
+
+The majority of these cases are necessary because they occur in an environment where service workers are not supported. The one exception to this is additional service workers and web workers. The HTTP requests for these workers are performed in a sandboxed environment within the browser that is not accessible from other service workers, so the Internet Computer service worker is unable to validate the responses to these requests and ensure that a replica has not responded with a malicious worker.
+
+Developers must be aware of each of these scenarios when developing a frontend dapp for the Internet Computer. This increases the amount of Internet Computer specific knowledge that developers need and by extension, the barrier of entry for developers into the ecosystem along with the likelihood of introducing unexpected bugs in dapps.
+
+#### Progressive Web Apps
+
+Service workers are scoped. Only one service worker may exist at a time on the same scope and a service worker may only intercept requests within its own scope. The Internet Computer Service Worker sits on the _root_ scope, so that it’s able to intercept all requests that are made by a dapp’s frontend. This means that dapps cannot have their own service worker on the root scope, so they are unable to provide features typical for [progressive web apps](https://web.dev/progressive-web-apps/), such as offline support, more aggressive caching strategies or background synchronization.
+
+#### Caching
+
+Browser caching is implemented based on response headers that are read by the browser before the request is intercepted by the service worker. Since these headers are not visible to the browser, browsers cannot effectively cache static assets as they would with normal web2 apps. Moving HTTP request translation to a remote server would mean that the requests are translated before reaching the browser, enabling browsers to implement caching as they normally would.
+
+With HTTP request translation being performed remotely on a server (as is the case with ICX Proxy), it will also be possible to provide proxy and CDN style caching. Bringing Internet Computer dapps even closer to realizing true web speed.
+
+#### Content Encoding
+
+[Brotli](https://en.wikipedia.org/wiki/Brotli#Applications) content encoding is superior to GZIP but is not currently supported by the service worker. This is because the browser would need to have visibility into the Content-Encoding header
```

Comment:
```suggestion
[Brotli](https://en.wikipedia.org/wiki/Brotli#Applications) content encoding is superior to GZIP but is not currently supported by the service worker. This is because the browser would need to have visibility into the Content-Encoding header and the Brotli encoded response body before it reaches the service worker in order to decode it correctly. Since the canister response body is hidden, along with the response headers, this is not possible, so the Service Worker needs to manually decode the response body.
```